### ISSUE TITLE
Rollup potentially overwriting unwanted field.

### ISSUE THREAD

**Issue by ghost** [afawcett/declarative-lookup-rollup-summaries#382]
Hi, I have encountered a situation where a rollup may potentially be overwriting an unwanted field.

I could write the specifics of it along with some screenshots, but given that this issue has been detailed already in Chatter, I thought just sharing the post link would suffice.

[https://success.salesforce.com/_ui/core/chatter/groups/GroupProfilePage?g=0F9300000009O5p&fId=0D53A00002o4yil&s1oid=00D300000000iTz&OpenCommentForEdit=1&emkind=chatterCommentNotification&s1nid=0DB30000000072L&emtm=1469803989468&s1uid=0053000000Bi5mV&fromEmail=1&s1ext=0](url)

Let me know if you need more details to troubleshoot it.

Thanks and again, kudos on DLRS!

Sebas

**Comment by afawcett** [comment]
Thanks i'll take a look! 👍

**Comment by afawcett** [comment]
Ah thanks @SebasCanseco, yeah i have just read that post. Can you have a look at my suggestion for next steps to debug this one?

**Comment by SebasCanseco** [comment]
I sent you a direct message on Chatter with more details. Please have a look. Thanks.

**Comment by sebascanseco** [comment]
Hi @afawcett. I was wondering if you have any updates regarding this topic?

**Comment by afawcett** [comment]
Community time has been a bit slow lately, due to me moving home/continent... i'm getting back into things though, and plan a spell of work on this tool soon. I have re categorized this issue as a priority enhancement. I suspect its also covered by another, but since i'm just aiming to get through all the outstanding GitHub issues right now, i'll leave this for now and connect it later....

**Comment by SebasCanseco** [comment]
I'm eager to know about your findings! I hope your moving has gone smoothly.

**Comment by afawcett** [comment]
Thanks, going well, will let you know on this asap.

**Comment by afawcett** [comment]
Hi @SebasCanseco, i am looking at this right now. I confess though, while i swear i saw your sandbox login cross my email box, i cannot find it now. Can you resend. I'll keep looking into this without in the meantime though today.

**Comment by afawcett** [comment]
Ok found the login... 👍

**Comment by afawcett** [comment]
Ok, it needs you @SebasCanseco to reset the password, i also sent you a message on Chatter. Thanks.

**Comment by afawcett** [comment]
Ok i have spent a couple of hours in your org getting to grips with your rollup configuration and I think i have found the source of the issue. 

If you look at **Setup > Custom Metadata > Rollup Summaries > Manage Records**. I created a new List View (which the main UI does not provide) to see things more clearly in terms of overall setup. 

![screen shot 2017-01-08 at 13 38 41](https://cloud.githubusercontent.com/assets/1167760/21753728/d4dbdb9a-d5a7-11e6-9e8e-904429a3e809.png)

The columns added are **Parent**, **Aggregate Result Field**, **Child** and **Field to Aggregate**. Highlighted below is that you have two active rollups that go into the same **Rating** field on **Account**. The one that is clearing your Rating field when adding Preferences under Account is **Wealth Engine Rating To Account Org**. 

![screen shot 2017-01-08 at 13 33 35](https://cloud.githubusercontent.com/assets/1167760/21753687/28f46f68-d5a7-11e6-8cbd-3b54a2f32f24.png)

Once i deactivate (unticked the Active tickbox) the **Wealth Engine Rating To Account Org** rollup things worked as expected. I suspect you may already know this, but are perhaps assuming that the criteria prevents overlap between this rollup and the one from Contacts. 

However the tool will be clearing the rollups target field when no records are matched by the criteria. This behaviour has been the source of much discussion, in terms of desired behaviour. For example in count scenarios, when the very last child record is deleted or drops out of the selected criteria, you would want the count to go to 0 and not stay at what the last value was. Likewise text fields are cleared and checkbox rollups are set to false.

In your case, i'm wondering if you would want more control over this, perhaps for the **Wealth Engine Rating To Account Org** rollup you don't care if there are no records found and prefer to leave whatever value is in the field intact? This still leaves the scenario in your case, of what to do if there is a value in it from the Contacts rollup? So i am really interested in your thoughts on what if anything the tool could offer to help out with this use case?

Also, keep in mind your use of TODAY criteria in realtime rollups. In this mode, the tool only evaluates your criteria when child records are inserted, updated or deleted. It will not evaluate it over the passage of time, and thus any condition you apply to things like TODAY have limited effect. This may not be an issue to you, if your just happy check at the time the record is inserted, updated or deleted. If it is an issue you will have to flip to full recalc scheduled of the rollup.

**Comment by SebasCanseco** [comment]
Andy thanks so much for spending time on this! I got to say though, I don't see you're mentioning the one rollup that - I believe - is causing the issue here. And that rollup is **Engagement_Score_To_Account**. 

If you read the first explanation of this issue in the Chatter group [post](https://success.salesforce.com/_ui/core/chatter/groups/GroupProfilePage?g=0F9300000009O5p&fId=0D53A00002o4yil&s1oid=00D300000000iTz&OpenCommentForEdit=1&emkind=chatterCommentNotification&s1nid=0DB30000000072L&emtm=1469803989468&s1uid=0053000000Bi5mV&fromEmail=1&s1ext=0) you can see that my testing got me to think that rollup is messing up with the calculated result of the rollup **Wealth_Engine_Rating_To_Account_Hou**, as in both rollups are storing the calculated value on the same object (Account) although different target fields.

If you, as a user, take actions to trigger the rollups in these specific order, you'll get that unexpected Rating field to be wiped:

1. From Preference__c.Type__c (Wealth Engine Rating) to Contact.Rating__c
2. From Contact.Rating__c (Wealth Engine Rating) to Account.Rating
3. From Preference__c.Value__c (Engagement Score) to Contact.Engagement_Score__c.
4. From Preference__c.Value__c (Engagement Score) to Account.Engagement_Score__c.

Or did I totally missed your actual resolution on this in your latest reply about this issue?

PS. so far having DLRS to set target fields as blank, when the are no records found, works pretty well for us.

**Comment by afawcett** [comment]
I did the tests, its for sure the '**Wealth Engine Rating To Account Org**' rollup. This and the other one, '**Wealth Engine Rating to Account Hou**' are both trying to rollup to the same **Account Rating** field. Disable (by unticking Active field) the **Wealth Engine Rating To Account Org** and you will see all is well.

**Comment by SebasCanseco** [comment]
Andy I have just joined another company but unfortunately I don't have access to the org we used for testing anymore. 

I'm still curious about how to get a workaround for this. I wonder if I didn't implement DLRS as I was supposed to. I'll try to replicate the issue sometime in the future and will come back to this thread and post any additional findings.

I truly appreciate your effort on getting to the bottom of this! Cheers!
